share/wake/lib/core/list.wake

```diff
@@ -114,58 +147,113 @@ export def mapPartial (f: a => Option b): List a => List b =
                 None   = sub
     loop
 
-# foldl: combine the elements of a List front-to-back into a single value.
+# Combine the elements of a List front-to-back into a single value.
 # An accumulator is updated from its initial value by combiningFn for each element.
 #
 # In C++ syntax, `foldl <combiningFn> <accumulateInitalValue> <List>` does:
+#   ```c++
 #   auto accumulator = <accumulateInitalValue>;
 #   for (element : <List>) accumulator = combiningFn(accumulator, element);
 #   return accumulator;
+#   ```
+# In python this is similar to `functools.reduce`
+#
+# Parameters:
+#  - `combiningFn`: The function to combine elements of the list with the accumulator
+#  - `initialValue`: The initial value of the accumulator
+#  - `list`: The list to combine the elements of
+#
+# Guarantees:
+#  - If `list` is Nil, the result is `initialValue`
```

Comment:
I understand that the 'golden example' included similarly a redundant `Guarantee`, but that was mostly because I was trying to show how to format stuff in  markdown. Since the examples already shows this claim, I think maybe we should not include the guarantee. There is a limit to how much text you get on a pop-up for documentation hover. I think we should try to include the most important stuff within that limit.